homeassistant/components/husqvarna_automower/lawn_mower.py

```diff
@@ -0,0 +1,124 @@
+
+    @property
+    def available(self) -> bool:
+        """Return True if the device is available."""
+        available = self.get_mower_attributes()["metadata"]["connected"]
+        return available
+
+    @property
+    def activity(self) -> LawnMowerActivity:
+        """Return the state of the mower."""
+        mower_attributes = AutomowerEntity.get_mower_attributes(self)
+        if mower_attributes["mower"]["state"] in [
+            "PAUSED",
+            "WAIT_UPDATING",
+            "WAIT_POWER_UP",
+        ]:
+            activity = LawnMowerActivity.PAUSED
+        if mower_attributes["mower"]["activity"] in ["MOWING", "LEAVING", "GOING_HOME"]:
+            activity = LawnMowerActivity.MOWING
+        if (mower_attributes["mower"]["state"] == "RESTRICTED") or (
+            mower_attributes["mower"]["activity"] in ["PARKED_IN_CS", "CHARGING"]
+        ):
+            activity = LawnMowerActivity.DOCKED
+        if (
+            mower_attributes["mower"]["state"]
+            in [
+                "FATAL_ERROR",
+                "ERROR",
+                "ERROR_AT_POWER_UP",
+                "NOT_APPLICABLE",
+                "UNKNOWN",
+                "STOPPED",
+                "OFF",
+            ]
+        ) or mower_attributes["mower"]["activity"] in ERROR_ACTIVITIES:
+            activity = LawnMowerActivity.ERROR
+        return LawnMowerActivity(activity)
+
+    @property
+    def extra_state_attributes(self) -> dict:
+        """Return the specific state attributes of this mower."""
+        mower_attributes = AutomowerEntity.get_mower_attributes(self)
+        action = mower_attributes["planner"]["override"]["action"]
+        action = action.lower() if action is not None else action
+        return {
+            "action": action,
+        }
+
+    async def async_start_mowing(self) -> None:
+        """Resume schedule."""
+        command_type = "actions"
+        payload = '{"data": {"type": "ResumeSchedule"}}'
+        try:
+            await self.coordinator.session.action(self.mower_id, payload, command_type)
+        except ClientResponseError as exception:
+            _LOGGER.error("Command couldn't be sent to the command que: %s", exception)
+
+    async def async_pause(self) -> None:
+        """Pauses the mower."""
+        command_type = "actions"
+        payload = '{"data": {"type": "Pause"}}'
+        try:
+            await self.coordinator.session.action(self.mower_id, payload, command_type)
+        except ClientResponseError as exception:
+            _LOGGER.error("Command couldn't be sent to the command que: %s", exception)
+
+    async def async_dock(self) -> None:
+        """Parks the mower until next schedule."""
+        command_type = "actions"
+        payload = '{"data": {"type": "ParkUntilNextSchedule"}}'
+        try:
+            await self.coordinator.session.action(self.mower_id, payload, command_type)
+        except ClientResponseError as exception:
```

Comment:
The library should preferably hide the transport details and raise library specific errors instead.

We shouldn't use the session directly as a library user. The library should provide a client that uses the transport session behind the scenes.